src/main/java/com/redhat/developers/msa/aloha/tracing/TracerResolver.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016 Juraci Paixão Kröhling
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.developers.msa.aloha.tracing;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.logging.Logger;
+
+import io.opentracing.Tracer;
+
+/**
+ * This is a helper class, that serves as an example of what could be done as a microservices library
+ * for detecting the appropriate Tracer based on the environment. On this case, we explicitly know about
+ * two tracers, BraveTracer and APMTracer.
+ *
+ * For the Hello World MSA, we might move this to a common module in the future.
+ *
+ * For real world applications, a code similar to this might be used, or a simplified version.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@SuppressWarnings("Duplicates")
+public class TracerResolver {
+    private static final Logger logger = Logger.getLogger(TracerResolver.class.getName());
+    private static Class<? extends Tracer> CACHED = null;
+
+    public static Tracer getTracer() {
+        if (CACHED != null) {
+            try {
+                return CACHED.newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                logger.warning(String.format("Failed to get instance from the cached class [%s]. Ignoring cache.", CACHED.getName()));
+            }
+        }
+
+        // first: explicit configuration via -Dtracer.class=fully.classified.class.Name
+        String tracerClassName = System.getProperty("tracer.class");
+        if (null == tracerClassName || tracerClassName.isEmpty()) {
+            // we didn't receive it as a sysprop, let's check the env
+            tracerClassName = System.getenv("TRACER_CLASS");
```

Comment:
This might be what we do in APM itself, but the usual practice is to have system properties as lowercase, with words separated by dots, like "tracer.class", whereas env vars are usually in all caps, works separated by "_".